lib/_emerge/depgraph.py

```diff
@@ -5255,6 +5260,25 @@ def _resolve(self, myfavorites):
                                 )
                             return 0, myfavorites
                         continue
+                    elif isinstance(arg, AtomArg):
+                        # If we're still an AtomArg at this point, it might mean we're
+                        # a candidate for pkgmove-fixing up.
+                        pkg = next(
+                            self._iter_match_pkgs_any(
+                                self._frozen_config.roots[myroot],
+                                atom,
+                                pkgmoves_only=True,
+                            ),
+                            None,
+                        )
+
+                        if pkg:
+                            arg = AtomArg(pkg.slot_atom)
+                            atom = pkg.slot_atom
+                            self._add_pkg(pkg, dep)
+                            self._create_graph()
```

Comment:
If _add_pkg or _create_graph returns 0 instead of 1 then you need to return. Normally _create_graph is delayed until later. after all the roots have been established.